src/js/components/history/HistoryItem.tsx

```diff
@@ -0,0 +1,192 @@
+  stopDragging: () => void;
+  setMarkVisual: (payload: {property: string, def: NumericValueRef | StringValueRef}, markId: number) => void;
+
+  bindChannel: (dsId: number, field: ColumnRecord, markId: number, property: string) => void;
+
+  setSignal: (value: SignalValue, signal: string) => void;
+}
+
+function mapDispatchToProps(dispatch: ThunkDispatch<State, null, AnyAction>, ownProps: OwnProps): DispatchProps {
+  return {
+    startDragging: (d: DraggingStateRecord) => {
+      dispatch(startDragging(d)); },
+    stopDragging: () => {
+      dispatch(stopDragging()); },
+    setMarkVisual: (p, markId) => {
+      dispatch(setMarkVisual(p, markId));
+    },
+    bindChannel: (dsId: number, field: ColumnRecord, markId: number, property: string) => {
+      dispatch(bindChannel(dsId, field, markId, property));
+    },
+    setSignal: (value: SignalValue, signal: string) => {
+      dispatch(setSignal(value, signal));
+    }
+  };
+}
+
+export class HistoryItemInspector extends React.Component<OwnProps & DispatchProps> {
+
+  constructor(props) {
+    super(props);
+  }
+
+  private width = 100; // these should match baseSignals in demonstrations.ts
+  private height = 100; //
+
+  public handleClick = (historyId: number) => {
+
+  }
+  public handleDragStart = (historyId: number) => {
+    this.props.startDragging(HistoryDraggingState({historyId}));
+
+    sg.set(MODE, 'channels');
+    ctrl.update();
+  }
+
+  public handleDragEnd = (evt: React.DragEvent<HTMLDivElement>, opts?) => {
+    const props = this.props;
+    const sel = sg.get(SELECTED);
+    const cell = sg.get(CELL);
+    const dropped = tupleid(sel) && tupleid(cell);
+
+    try {
+      const lyraId = +sel.mark.role.split('lyra_')[1]; // id of thing that was dropped onto
+
+      if (dropped) {
+        const channel = channelName(cell.key);
+        let fieldName, dsId;
+        if (channel === 'x' || channel === 'y' || channel === 'color' || channel === 'size') {
+          // set scale
+          let channelScaleIds = this.props.history.getIn(["guides"])
+            .map((g) => {
+              return channel == 'x' || channel == 'y' ? g.scale : channel == 'size' ? g[channel] : g.fill;
+            })
+            .filter((scaleId) => {
+              let scaleName = scaleId ? this.props.history.getIn(["scales"]).get(scaleId).name : null;
+              return scaleName === channel;
+            });
+
+          fieldName = channelScaleIds.map((scaleId) => {
+            let scaleRecord = this.props.history.getIn(["scales"]).get(scaleId);
+            return scaleRecord.get('_domain').length > 0 ?  scaleRecord.get('_domain')[0].field : null;
+          }).first();
+
+          dsId = channelScaleIds.map((scaleId) => {
+            let scaleRecord = this.props.history.getIn(["scales"]).get(scaleId);
+            return scaleRecord.get('_domain').length > 0 ?  scaleRecord.get('_domain')[0].data : null;
+          }).first();
+
+
+        }
```

Comment:
this is fine for now as you're prototyping, but from a design perspective I don't think we should use dropping outside of a dropzone as the final interaction for non-x/y channels

Reply:
+1